src/patch-prototypes.js

```diff
@@ -0,0 +1,94 @@
+import {HTMLElementPatches} from './patches/HTMLElement.js';
+import {SlotPatches} from './patches/Slot.js';
+import {DocumentOrFragmentPatches} from './patches/DocumentOrFragment.js';
+import {DocumentOrShadowRootPatches} from './patches/DocumentOrShadowRoot.js';
+import {DocumentPatches} from './patches/Document.js';
+import {WindowPatches} from './patches/Window.js';
+
+// Some browsers (IE/Edge) have non-standard HTMLElement accessors.
+const NonStandardHTMLElement = {};
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'parentElement')) {
+  Object.defineProperty(NonStandardHTMLElement, 'parentElement',
+    Object.getOwnPropertyDescriptor(NodePatches, 'parentElement'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'contains')) {
+  Object.defineProperty(NonStandardHTMLElement, 'contains',
+    Object.getOwnPropertyDescriptor(NodePatches, 'contains'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'children')) {
+  Object.defineProperty(NonStandardHTMLElement, 'children',
+    Object.getOwnPropertyDescriptor(ParentNodePatches, 'children'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'innerHTML')) {
+  Object.defineProperty(NonStandardHTMLElement, 'innerHTML',
+    Object.getOwnPropertyDescriptor(ElementOrShadowRootPatches, 'innerHTML'));
+}
+
+// setup patching
+const patchMap = {
+  EventTarget: [EventTargetPatches],
+  Node: [NodePatches, !window.EventTarget ? EventTargetPatches : null],
+  Text: [SlotablePatches],
+  Element: [ElementPatches, ParentNodePatches, SlotablePatches,
+    utils.settings.IS_IE ? null : ElementOrShadowRootPatches,
+    !window.HTMLSlotElement ? SlotPatches : null],
+  HTMLElement: [HTMLElementPatches, NonStandardHTMLElement],
+  HTMLSlotElement: [SlotPatches],
+  DocumentFragment: [ParentNodeDocumentOrFragmentPatches, DocumentOrFragmentPatches],
+  Document: [DocumentPatches, ParentNodeDocumentOrFragmentPatches, DocumentOrFragmentPatches, DocumentOrShadowRootPatches],
+  Window: [WindowPatches]
+}
+
+const getPatchPrototype = (name) => (name === 'HTMLElement') ?
+    utils.NativeHTMLElement.prototype :
+    window[name] && window[name].prototype;
+
+const onPatchMap = (fn) => {
+  for (let p in patchMap) {
+    const proto = getPatchPrototype(p);
+    patchMap[p].forEach(patch => proto && patch && fn(proto, patch));
+  }
+}
+
+export const addShadyPrefixedProperties = () => {
+  // perform shady patches
+  onPatchMap((proto, patch) =>
+    utils.patchProperties(proto,
+      utils.getOwnPropertyDescriptors(patch), true, utils.SHADY_PREFIX));
+
+  // install `_activeElement` because some browsers (older Chrome/Safari) do not have
+  // a 'configurable' `activeElement` accesssor.
+  const descriptor = Object.getOwnPropertyDescriptor(DocumentOrShadowRootPatches, 'activeElement');
+  Object.defineProperty(Document.prototype, '_activeElement', descriptor);
```

Comment:
Patch document instance instead, rather than use a non-standard name?

Reply:
Patched standard name in addition on `document` instance.